File: map_software/src/storage/jsonParser.js
const fs = require('fs');

const inputFileName = 'snapshot_new.json';
const outputFileName = 'output.json';

const transformData = (inputData) => {
    let outputData = [];
    inputData.IAS.System.networks.forEach((network, i) => {
        if ('IMT' in network.network) {
            const bsData = network.network.IMT.deployment.BS;
            const ueData = network.network.IMT.UE;

            bsData.forEach((bs, i) => {
                const baseStationId = `BS_${i}`;
                const bsLat = bs.location.latitude;
                const bsLong = bs.location.longitude;

                const ueList = ueData.filter(ue => ue.base_station_id === i);
                const ueOutput = [];

                ueList.forEach((ue, j) => {
                    ueOutput.push({
                        "UE_ID": `UE_${i}_${j}`,
                        "Latitude": ue.location.latitude,
                        "Longitude": ue.location.longitude
                    });
                });

                outputData.push({
                    "Base_Station_ID": baseStationId,
                    "Latitude": bsLat,
                    "Longitude": bsLong,
                    "UEs": ueOutput
                });
            });
        }
    });
    return outputData;
};

fs.readFile(inputFileName, 'utf8', (err, data) => {
    if (err) {
        console.error('Error reading the file:', err);
        return;
    }
    const inputJson = JSON.parse(data);
    const transformedData = transformData(inputJson);

    fs.writeFile(outputFileName, JSON.stringify(transformedData, null, 4), (err) => {
        if (err) {
            console.error('Error writing the file:', err);
            return;
        }
        console.log(`Data transformed and saved to ${outputFileName}`);
    });
});

File: map_software/src/storage/jsonParser.py
import json

input_file = 'snapshot_new.json'
output_file = 'output.json'

def transform_data(input_data):
    output_data = []
    for network in input_data['IAS']['System']['networks']:
        if 'IMT' in network['network']:
            bs_data = network['network']['IMT']['deployment']['BS']
            ue_data = network['network']['IMT']['UE']
            
            for i, bs in enumerate(bs_data):
                base_station_id = f"BS_{i}"
                bs_lat = bs['location']['latitude']
                bs_long = bs['location']['longitude']
                
                ue_list = [ue for ue in ue_data if ue['base_station_id'] == i]
                ue_output = []
                for j, ue in enumerate(ue_list):
                    ue_output.append({
                        "UE_ID": f"UE_{i}_{j}",
                        "Latitude": ue['location']['latitude'],  
                        "Longitude": ue['location']['longitude'] 
                    })
                
                output_data.append({
                    "Base_Station_ID": base_station_id,
                    "Latitude": bs_lat,
                    "Longitude": bs_long,
                    "UEs": ue_output
                })
    
    return output_data

with open(input_file, 'r') as file:
    input_json = json.load(file)

transformed_data = transform_data(input_json)

with open(output_file, 'w') as file:
    json.dump(transformed_data, file, indent=4)

print(f"Data transformed and saved to {output_file}")
